File: Sample/sample_cpp/Graphics/2D/ImagePackage.cpp

#include <Base.h>

/**
@brief	エフェクトを表示する。
*/
void ImagePackage()
{
	// aceを初期化する
	asd::Engine::Initialize(asd::ToAString("ImagePackage").c_str(), 640, 480, asd::EngineOption());

	// シーンを生成する
	auto scene = std::make_shared<asd::Scene>();

	// レイヤーを生成する
	auto layer = std::make_shared<asd::Layer2D>();

	// シーンにレイヤーを追加する
	scene->AddLayer(layer);

	// シーンを切り替える
	asd::Engine::ChangeScene(scene);

	// イメージパッケージを読み込む
	auto imagePackage = asd::Engine::GetGraphics()->CreateImagePackage(asd::ToAString("Data/ImagePackage/Game.aip").c_str());

	for (int i = 0; i < imagePackage->GetImageCount(); i++)
	{
		// テクスチャを取り出す
		auto texture = imagePackage->GetImage(i);
		auto area = imagePackage->GetImageArea(i);

		// テクスチャをオブジェクトとして追加する
		auto textureObject2D = std::make_shared<asd::TextureObject2D>();
		textureObject2D->SetTexture(texture);
		textureObject2D->SetPosition(asd::Vector2DF(area.X, area.Y));
		layer->AddObject(textureObject2D);
	}

	// aceが進行可能かチェックする。
	while (asd::Engine::DoEvents())
	{
		// aceを更新する。
		asd::Engine::Update();
	}

	// aceを終了する。
	asd::Engine::Terminate();
}

File: Sample/BasicSample/sample_cs/Graphics/2D/ImagePackage.cs

/// <summary>
/// ImagePackageのサンプル。画像パッケージを元に画像を配置しています。
/// </summary>
class ImagePackage : ISample
{
	public void Run()
	{
		// AC-Engineを初期化する。
		ace.Engine.Initialize("ImagePackage", 640, 480, new ace.EngineOption());

		// シーンを生成する
		var scene = new ace.Scene();

		// レイヤーを生成する
		var layer = new ace.Layer2D();

		// シーンにレイヤーを追加する
		scene.AddLayer(layer);

		// シーンを切り替える
		ace.Engine.ChangeScene(scene);

		// イメージパッケージを読み込む
		ace.ImagePackage imagePackage = ace.Engine.Graphics.CreateImagePackage("Data/ImagePackage/Game.aip");

		for(int i = 0; i < imagePackage.ImageCount; i++)
		{
			// テクスチャを取り出す
			ace.Texture2D texture = imagePackage.GetImage(i);
			ace.RectI area = imagePackage.GetImageArea(i);

			// テクスチャをオブジェクトとして追加する
			ace.TextureObject2D textureObject2D = new ace.TextureObject2D();
			textureObject2D.Texture = texture;
			textureObject2D.Position = new ace.Vector2DF(area.X, area.Y);
			layer.AddObject(textureObject2D);
		}

		// AC-Engineのウインドウが閉じられていないか確認する。
		while (ace.Engine.DoEvents())
		{
			// AC-Engineを更新する。
			ace.Engine.Update();
		}

		// AC-Engineの終了処理をする。
		ace.Engine.Terminate();
	}
}

